crates/zkvm/zkvm/src/proof.rs

```diff
@@ -2,80 +2,98 @@ use arbitrary::Arbitrary;
 use borsh::{BorshDeserialize, BorshSerialize};
 use serde::{Deserialize, Serialize};
 
-/// Validity proof generated by the `ZkVmHost`
+/// Represents a wrapper around a byte vector for various proofs and keys.
+///
+/// Provides common utilities such as byte access and emptiness checks.
 #[derive(
     Debug, Clone, Serialize, Deserialize, BorshSerialize, BorshDeserialize, PartialEq, Eq, Arbitrary,
 )]
-pub struct Proof(Vec<u8>);
+pub struct ByteWrapper(Vec<u8>);
```

Comment:
Maybe a better name for this is like `ProverBuf` or something?  Why can't we just be using `Vec<u8>` directly in all these places?

Reply:
Yes, `ProofBuf` makes sense. `ByteWrapper` has been introduced, because we were exposing same functions for the `Proof`, `PublicValues` and the `VerificationKey` all of which have a different type for different ZkVm.